source/math/FloatPoint2DVectorList.ooc

```diff
@@ -69,6 +69,22 @@ FloatPoint2DVectorList: class extends VectorList<FloatPoint2D> {
 		sortedY free()
 		result
 	}
+	getMean: func (indices: VectorList<Int>) -> FloatPoint2D {
+		result := FloatPoint2D new()
+		if (indices count > 0) {
+			buffer := this pointer as FloatPoint2D*
+			indicesBuffer := indices pointer as Int*
+			for (i in 0 .. indices count) {
```

Comment:
It's not an expensive function, but since you mentioned this being function being a bottleneck...